File: remote/src/main/java/com/kelsos/mbrc/annotations/PlayerState.java
package com.kelsos.mbrc.annotations;

import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class PlayerState {
  public static final String PLAYING = "playing";
  public static final String PAUSED = "paused";
  public static final String STOPPED = "stopped";

  private PlayerState() {
    //no instance
  }

  @Retention(RetentionPolicy.SOURCE)
  @StringDef({
                 PAUSED,
                 PLAYING,
                 STOPPED
             })
  public @interface State {
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/content/active_status/PlayerState.kt
package com.kelsos.mbrc.content.active_status

import androidx.annotation.StringDef

object PlayerState {
  const val PLAYING = "playing"
  const val PAUSED = "paused"
  const val STOPPED = "stopped"
  const val UNDEFINED = "undefined"

  @Retention(AnnotationRetention.SOURCE)
  @StringDef(PAUSED, PLAYING, STOPPED, UNDEFINED)
  annotation class State
} // no instance
